File: app/src/main/java/com/example/solar_alarm/CreateAlarm/CreateAlarmViewModel.java
package com.example.solar_alarm.CreateAlarm;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;

import com.example.solar_alarm.Data.Alarm;
import com.example.solar_alarm.Data.AlarmRepository;

public class CreateAlarmViewModel extends AndroidViewModel {
    private AlarmRepository alarmRepository;

    public CreateAlarmViewModel(@NonNull Application application) {
        super(application);

        alarmRepository = new AlarmRepository(application);
    }

    public void insert(Alarm alarm) {
        alarmRepository.insert(alarm);
    }

    public void update(Alarm alarm) { alarmRepository.update(alarm); }
}

File: app/src/main/java/com/example/solar_alarm/CreateAlarm/CreateAlarmViewModel.kt
package com.example.solar_alarm.CreateAlarm

import com.example.solar_alarm.Data.Alarm
import android.app.Application
import android.os.Build
import androidx.annotation.RequiresApi
import androidx.lifecycle.AndroidViewModel
import com.example.solar_alarm.Data.AlarmRepository

@RequiresApi(Build.VERSION_CODES.O)
class CreateAlarmViewModel(application: Application) : AndroidViewModel(application) {
    private val alarmRepository: AlarmRepository

    init {
        alarmRepository = AlarmRepository(application)
    }

    fun insert(alarm: Alarm?) {
        alarmRepository.insert(alarm)
    }

    fun update(alarm: Alarm?) {
        alarmRepository.update(alarm)
    }
}